config/app/com/microsoft/azure/iotsolutions/uiconfig/services/runtime/KeyVault.java

```diff
@@ -0,0 +1,147 @@
+import play.Logger;
+
+import java.net.MalformedURLException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+
+public class KeyVault {
+
+    // Key Vault details and access
+    private final String name;
+    private final String clientId;
+    private final String clientSecret;
+
+    private static final Logger.ALogger log = Logger.of(KeyVault.class);
+
+    // Key Vault Client
+    private final KeyVaultClient keyVaultClient;
+    private List<SecretItem> keys;
+
+    // Constants
+    private final static String KEY_VAULT_URI = "https://%s.vault.azure.net/secrets/%s";
+
+    public KeyVault(String name, String clientId, String clientSecret) {
+        this.name = name;
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.keyVaultClient = new KeyVaultClient(createCredentials());
+        this.getAllKeys();
+    }
+
+    public String getKeyVaultSecret(String secretKey) {
+        secretKey = this.processSecretKey(secretKey);
+        String uri = String.format(KEY_VAULT_URI, this.name, secretKey);
+
+        try {
+            return this.keyVaultClient
+                    .getSecret(uri)
+                    .value();
+        } catch (Exception e) {
+            String message = String.format("Failed to get the secret {%s} from the key vault.", secretKey);
+            log.error(message, e);
+            return null;
+        }
+    }
+
+    public boolean hasPath(String key) {
+        return this.keys.stream()
+                .anyMatch(
+                        item -> item
+                                .id()
+                                .toLowerCase()
+                                .contains(key.toLowerCase())
+                );
+    }
+
+    // Get last token of the key for referencing the key vault:
+    // (local setting file ) => key vault denomination
+    // messages.cosmosdb.documentDbConnectionString => documentDbConnectionString
+    private String processSecretKey(String secretKey) {
+        return secretKey.substring(secretKey.lastIndexOf("."));
+    }
+
+    /**
+     * Creates a new KeyVaultCredential based on the access token obtained.
+     *
+     * @return
+     */
+    private ServiceClientCredentials createCredentials() {
+        return new KeyVaultCredentials() {
+
+            //Callback that supplies the token type and access token on request.
+            @Override
+            public String doAuthenticate(String authorization, String resource, String scope) {
+
+                AuthenticationResult authResult;
+                try {
+                    authResult = getAccessToken(authorization, resource);
+                    return authResult.getAccessToken();
+                } catch (Exception e) {
+                    // TODO: Add logging
+                    // e.printStackTrace();
```

Comment:
Add logging now.

Reply:
added

Reply:
This function is similar to examples from Keyvault library. It seems we need to return "" string than null value.